core/chains/evm/gas/chainoracles/l1_gas_price_oracle.go

```diff
@@ -0,0 +1,176 @@
+package chainoracles
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/exp/slices"
+
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+//go:generate mockery --quiet --name ethClient --output ./mocks/ --case=underscore --structname ETHClient
+type ethClient interface {
+	CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
+}
+
+// Reads L2-specific precompiles and caches the l1GasPrice set by the L2.
+type l1GasPriceOracle struct {
+	client     ethClient
+	pollPeriod time.Duration
+	logger     logger.Logger
+	address    string
+	callArgs   string
+
+	l1GasPriceMu sync.RWMutex
+	l1GasPrice   *assets.Wei
+
+	chInitialised chan struct{}
+	chStop        utils.StopChan
+	chDone        chan struct{}
+	utils.StartStopOnce
+}
+
+const (
+	// ArbGasInfoAddress is the address of the "Precompiled contract that exists in every Arbitrum chain."
+	// https://github.com/OffchainLabs/nitro/blob/f7645453cfc77bf3e3644ea1ac031eff629df325/contracts/src/precompiles/ArbGasInfo.sol
+	ArbGasInfoAddress = "0x000000000000000000000000000000000000006C"
+	// ArbGasInfo_getL1BaseFeeEstimate is the a hex encoded call to:
+	// `function getL1BaseFeeEstimate() external view returns (uint256);`
+	ArbGasInfo_getL1BaseFeeEstimate = "f5d6ded7"
+
+	// GasOracleAddress is the address of the precompiled contract that exists on OP stack chain.
+	// This is the case for Optimism and Base.
+	OPGasOracleAddress = "0x420000000000000000000000000000000000000F"
+	// GasOracle_l1BaseFee is the a hex encoded call to:
+	// `function l1BaseFee() external view returns (uint256);`
+	OPGasOracle_l1BaseFee = "519b4bd3"
+
+	// Interval at which to poll for L1BaseFee. A good starting point is the L1 block time.
+	PollPeriod = 12 * time.Second
+)
+
+var supportedChainTypes = []config.ChainType{config.ChainArbitrum, config.ChainOptimismBedrock}
+
+func IsL1OracleChain(chainType config.ChainType) bool {
```

Comment:
This name can confuse someone into thinking if this chain itself is an L1 chain.
Is this a better name: IsRollupWithL1Support.